WalletWasabi.Gui/Controls/WalletExplorer/BuildTabViewModel.cs

```diff
@@ -15,18 +16,21 @@ public class BuildTabViewModel : SendControlViewModel
 		public override string DoButtonText => "Build Transaction";
 		public override string DoingButtonText => "Building Transaction...";
 
-		public BuildTabViewModel(WalletViewModel walletViewModel) : base(walletViewModel, "Build Transaction")
+		private Guid _walletId;
+
+		public BuildTabViewModel(WalletService walletService, Guid walletId) : base(walletService, "Build Transaction")
```

Comment:
Why only the `BuildTabViewModel` has a `Guid walletId` parameter?

Reply:
For some reason it needed to identify by Id, and just because of this requirement every ViewModel was having to be passed WalletViewModel which is a bit much, so now im only passing Guid to the single model that needs it.

Reply:
P.S. im thinking of making this change a seperate PR.

Reply:
Actually I cant easily, because it depends alot of other changes here.

Im thinking to get rid of the ID and making WalletService comparable. then you can safely identify the wallet without creating a random guid.

Reply:
Obsoleted the use of Ids, it does this now by comparing WalletServices.